src/de/mod.rs

```diff
@@ -1,6 +1,1324 @@
+//! to the name of attributes.
+//! </div>
+//! </td>
+//! </tr>
+//! <!-- 5 ===================================================================================== -->
+//! <tr>
+//! <td>
+//! An XML with an attribute and a child element named equally:
+//!
+//! ```xml
+//! <any-tag field="...">
+//!   <field>...</field>
+//! </any-tag>
+//! ```
+//! </td>
+//! <td>
+//!
+//! You MUST specify `#[serde(rename = "@field")]` on a field that will be used
+//! for an attribute.
+//!
+//! ```
+//! # use pretty_assertions::assert_eq;
+//! # use serde::Deserialize;
+//! # type T = ();
+//! # type U = ();
+//! # #[derive(Debug, PartialEq)]
+//! #[derive(Deserialize)]
+//! struct AnyName {
+//!   #[serde(rename = "@field")]
+//!   attribute: T,
+//!   field: U,
+//! }
+//! # assert_eq!(
+//! #   AnyName { attribute: (), field: () },
+//! #   quick_xml::de::from_str(r#"
+//! #     <any-tag field="...">
+//! #       <field>...</field>
+//! #     </any-tag>
+//! #   "#).unwrap(),
+//! # );
+//! ```
+//! </td>
+//! </tr>
+//! <!-- ======================================================================================= -->
+//! <tr><th colspan="2">
+//!
+//! ## Optional attributes and elements
+//!
+//! </th></tr>
+//! <tr><th>To parse all these XML's...</th><th>...use that Rust type(s)</th></tr>
+//! <!-- 6 ===================================================================================== -->
+//! <tr>
+//! <td>
+//! An optional XML attribute that you want to capture.
+//! The root tag name does not matter.
+//!
+//! ```xml
+//! <any-tag optional="..."/>
+//! ```
+//! ```xml
+//! <any-tag/>
+//! ```
+//! </td>
+//! <td>
+//!
+//! A structure with an optional field, renamed according to the requirements
+//! for attributes:
+//!
+//! ```
+//! # use pretty_assertions::assert_eq;
+//! # use serde::Deserialize;
+//! # type T = ();
+//! # #[derive(Debug, PartialEq)]
+//! #[derive(Deserialize)]
+//! struct AnyName {
+//!   #[serde(rename = "@optional")]
+//!   optional: Option<T>,
+//! }
+//! # assert_eq!(AnyName { optional: Some(()) }, quick_xml::de::from_str(r#"<any-tag optional="..."/>"#).unwrap());
+//! # assert_eq!(AnyName { optional: None     }, quick_xml::de::from_str(r#"<any-tag/>"#).unwrap());
+//! ```
+//! When the XML attribute is present, type `T` will be deserialized from
+//! an attribute value (which is a string). Note, that if `T = String` or other
+//! string type, the empty attribute is mapped to a `Some("")`, whereas `None`
+//! represents the missed attribute.
+//! </td>
+//! </tr>
+//! <!-- 7 ===================================================================================== -->
+//! <tr>
+//! <td>
+//! An optional XML elements that you want to capture.
+//! The root tag name does not matter.
+//!
+//! ```xml
+//! <any-tag/>
+//!   <optional>...</optional>
+//! </any-tag>
+//! ```
+//! ```xml
+//! <any-tag/>
+//! ```
+//! </td>
+//! <td>
+//!
+//! A structure with an optional field:
```

Comment:
This might be a good place to point out some of the known peculiarities, e.g. https://github.com/tafia/quick-xml/issues/497

Reply:
Added a known bug mark